content/blog/2020-01-ggplot2-3-3-0/index.Rmarkdown

```diff
@@ -0,0 +1,231 @@
+
+# Overlapping labels
+plot(p)
+```
+
+```{r}
+# Use guide_axis to dodge the labels
+p + 
+  scale_x_discrete(guide = guide_axis(n.dodge = 2))
+```
+
+```{r}
+# Or to remove overlapping labels
+p + 
+  scale_x_discrete(guide = guide_axis(check.overlap = TRUE))
+```
+
+### New bin scale
+While ggplot2 has a lot of different scales, it has only ever had two different scale types: *continuous* and *discrete*. This all changes with this release, as a new binning scale type has been added. A binning scale takes continuous data and makes it discrete by assigning each data point to a bin. It thus exist somewhere between the two existing scale types and e.g. allow continuous data to be used with discrete palettes. Being a fundamental scale type it is available to all aesthetics, both positional and otherwise:
+
+```{r}
+p <- ggplot(mpg) + 
+  geom_point(aes(displ, cty, size = hwy, colour = hwy))
+
+p + 
+  scale_size_binned()
+```
+
+```{r}
+p + 
+  scale_colour_binned()
+```
+
+As can be seen, the legends will reflect the binned nature of the scale by placing tick marks at the border between bins. By default the out-most ticks are not shown, indicating that the binning is open in both ends. This behavior can be controlled in the guide.
+
+```{r}
+p + 
+  scale_size_binned(guide = guide_bins(show.limits = TRUE))
+```
+
+When used with a positional scale it acts in much the same way, placing data at the center of the bin they belong to:
+
+```{r}
+p + 
+  scale_x_binned()
+```
+
+One of the benefits of this is that it is now a breeze to create histograms with tick-marks between the bars, as a histogram is effectively a binned scale with a bar geom:
+
+```{r}
+ggplot(mpg) + 
+  geom_bar(aes(displ)) + 
+  scale_x_binned()
+```
+
+### Bi-directional geoms and stats
+While we are messing with the foundation of ggplot2, we might as well challenge another paradigm, namely that certain geoms have a direction, and to change the direction you'd have to use `coord_flip()`. The prime example of this is a horizontal bar chart:
+
+```{r}
+ggplot(mpg) + 
+  geom_bar(aes(x = manufacturer)) + 
+  coord_flip()
+```
+
+This approach has served ggplot2 well, and will continue to work in the future, but with this release we update all the directional stats and geoms to work in both directions. The direction of the stat/geom is deduced from the aesthetic mapping, so it should simply behave as expected. The example above can thus be rewritten to:
+
+```{r}
+ggplot(mpg) + 
+  geom_bar(aes(y = manufacturer))
+```
+
+The direction determination is not only looking at which aesthetics are mapped, but also what they are mapped to, so most layers can be determined without ambiguity. In the presence of ambiguity it is conservative and will default to the standard orientation. If you want to overwrite the direction, either because it fails to detect it, or because the geom/stat is ambiguous by nature, It can be set directly with the `orientation` argument:
```

Comment:
```suggestion
The direction determination is not only looking at which aesthetics are mapped, but also what they are mapped to, so most layers can be determined without ambiguity. In the presence of ambiguity, it is conservative and will default to the standard orientation. If you want to overwrite the direction, either because it fails to detect it, or because the geom/stat is ambiguous by nature, it can be set directly with the `orientation` argument:
```

Reply:
Conditional comma, and a renegade capital